java/arcs/sdk/Particle.kt

```diff
@@ -11,8 +11,33 @@
 
 package arcs.sdk
 
+/**
+ * Interface used by [ArcHost]s to interact dynamically with code-generated [Handle] fields
+ * used by [Particle]s.
+ */
+interface HandleHolder {
+    val map: MutableMap<String, Handle>
```

Comment:
Nit: say something about what goes in the map. Either `handleMap` or just `handles` would be fine.